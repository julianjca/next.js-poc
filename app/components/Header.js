import Link from 'next/link'

const linkStyle = {
  marginRight: 15
}

const Header = (props) => (
  <div>
    <h2>{props.title}</h2>
    <hr />
    <Link href="/">
      <a style={linkStyle}>Home</a>
    </Link>
    <Link href="/about">
      <a style={linkStyle}>About</a>
    </Link>
  </div>
)

export default Header