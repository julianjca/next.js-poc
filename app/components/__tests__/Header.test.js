import renderer  from 'react-test-renderer'
import shallowRenderer from 'react-test-renderer/shallow'
import Header from '../Header'

it('Header should show correct', () => {
  const tree = renderer.create(
    <Header title="Hello" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
});

it('Header should show correct (shallow)', () => {
  const renderer = new shallowRenderer
  const snapshot = renderer.render(
    <Header title="Hello" />
  )
  expect(snapshot).toMatchSnapshot()
});
