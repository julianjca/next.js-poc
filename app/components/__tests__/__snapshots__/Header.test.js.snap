// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should show correct (shallow) 1`] = `
<div>
  <h2>
    Hello
  </h2>
  <hr />
  <Link
    href="/"
  >
    <a
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Home
    </a>
  </Link>
  <Link
    href="/about"
  >
    <a
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      About
    </a>
  </Link>
</div>
`;

exports[`Header should show correct 1`] = `
<div>
  <h2>
    Hello
  </h2>
  <hr />
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "marginRight": 15,
      }
    }
  >
    Home
  </a>
  <a
    href="/about"
    onClick={[Function]}
    style={
      Object {
        "marginRight": 15,
      }
    }
  >
    About
  </a>
</div>
`;
